sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -352,8 +352,9 @@ case class AdaptiveSparkPlanExec(
           val newCost = costEvaluator.evaluateCost(newPhysicalPlan)
           if (newCost < origCost ||
             (newCost == origCost && currentPhysicalPlan != newPhysicalPlan)) {
-            logOnLevel("Plan changed:\n" +
-              sideBySide(currentPhysicalPlan.treeString, newPhysicalPlan.treeString).mkString("\n"))
+            lazy val plans = sideBySide(
+              currentPhysicalPlan.treeString, newPhysicalPlan.treeString).mkString("\n")
+            logOnLevel("Plan changed:\n" + plans)
```

Comment:
if `logOnLevel` eagerly evaluates its arguments, this change won't help, right?

Reply:
yes

Reply:
but if it's not eager, the previous code is also fine?

Reply:
user can change `spark.sql.adaptive.logLevel` to warn or we can change the default value to warn

Reply:
it's not related to the log level. The string content passed to `logOnLevel` is lazily evaluated. How can it cause OOM if the log is not written eventually?

Reply:
Sorry, It was my mistake, I added this configuration myself:
```
logger.adaptive.name = org.apache.spark.sql.execution.adaptive
logger.adaptive.level = debug
```